crates/typst-pdf/src/embed.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{PdfChunk, WithGlobalRefs};
+use ecow::EcoString;
+use pdf_writer::{Date, Finish, Name, Ref, Str, TextStr};
+use std::collections::HashMap;
+use typst::diag::SourceResult;
+
+pub fn write_embedded_files(
+    ctx: &WithGlobalRefs,
+) -> SourceResult<(PdfChunk, HashMap<EcoString, Ref>)> {
+    let mut chunk = PdfChunk::new();
+    let mut embedded_files = HashMap::default();
+    for embed in &ctx.resources.embeds {
+        let embedded_file_stream_ref = chunk.alloc.bump();
+        let file_spec_dict_ref = chunk.alloc.bump();
+
+        let length = embed.data().len();
+        let mut embedded_file =
+            chunk.embedded_file(embedded_file_stream_ref, embed.data().as_ref());
+        embedded_file
+            .subtype(Name(b"text/xml"))
+            .pair(Name(b"Length"), length as i32);
+        embedded_file.params().modification_date(Date::new(2023)).finish(); // Todo: can we just let this out?
```

Comment:
Maybe use the document creation date here; if we have one?

Reply:
We do create a document date somewhere, check out the code where we write the XMP metadata.